src/include/byteorder.h

```diff
@@ -67,11 +82,9 @@ MAKE_LE_CLASS(32)
 MAKE_LE_CLASS(16)
 #undef MAKE_LE_CLASS
 
-#endif /* __cplusplus */
-
-#define init_le64(x) { (__u64)mswab64(x) }
-#define init_le32(x) { (__u32)mswab32(x) }
-#define init_le16(x) { (__u16)mswab16(x) }
+#define init_le64(x) { (__u64)mswab(x) }
+#define init_le32(x) { (__u32)mswab(x) }
+#define init_le16(x) { (__u16)mswab(x) }
```

Comment:
seems like this needs the template <> parameter since we don't know that x has the right type